File: src/main/java/com/windea/java/utils/JsonYamlUtils.java
package com.windea.java.utils;

public class JsonYamlUtils {
	private JsonYamlUtils() {
	}


	public static void jsonToYamlFile(String path) throws Exception {
		jsonToYamlFile(path, PathUtils.changeFileExt(path, ".yml"));
	}

	public static void jsonToYamlFile(String jsonPath, String yamlPath) throws Exception {
		var data = JsonUtils.fromFile(jsonPath);
		YamlUtils.toFile(data, yamlPath);
	}

	public static void yamlToJsonFile(String path) throws Exception {
		yamlToJsonFile(path, PathUtils.changeFileExt(path, ".json"));
	}

	public static void yamlToJsonFile(String yamlPath, String jsonPath) throws Exception {
		var data = YamlUtils.fromFile(yamlPath);
		JsonUtils.toFile(data, jsonPath);
	}
}

File: src/main/kotlin/com/windea/kotlin/utils/JsonYamlUtils.kt
package com.windea.kotlin.utils

object JsonYamlUtils {
	@Throws(Exception::class)
	fun jsonToYamlFile(path: String) {
		jsonToYamlFile(path, PathUtils.changeFileExt(path, ".yml"))
	}

	@Throws(Exception::class)
	fun jsonToYamlFile(jsonPath: String, yamlPath: String) {
		val data = JsonUtils.fromFile(jsonPath)
		YamlUtils.toFile(data, yamlPath)
	}

	@Throws(Exception::class)
	fun yamlToJsonFile(path: String) {
		yamlToJsonFile(path, PathUtils.changeFileExt(path, ".json"))
	}

	@Throws(Exception::class)
	fun yamlToJsonFile(yamlPath: String, jsonPath: String) {
		val data = YamlUtils.fromFile(yamlPath)
		JsonUtils.toFile(data, jsonPath)
	}
}
